src/nwb/RecordingContainers.hpp

```diff
@@ -0,0 +1,92 @@
+#pragma once
+
+#include "Channel.hpp"
+#include "Types.hpp"
+#include "nwb/base/TimeSeries.hpp"
+
+namespace AQNWB::NWB
+{
+
+/**
+ * @brief The RecordingContainers class provides an interface for managing
+ * and holding groups of Containers acquired during a recording.
+ */
+
+class RecordingContainers
+{
+public:
+  /**
+   * @brief Constructor for RecordingContainer class.
+   */
+  RecordingContainers();
+
+  /**
+   * @brief Deleted copy constructor to prevent construction-copying.
+   */
+  RecordingContainers(const RecordingContainers&) = delete;
+
+  /**
+   * @brief Deleted copy assignment operator to prevent copying.
+   */
+  RecordingContainers& operator=(const RecordingContainers&) = delete;
+
+  /**
+   * @brief Destructor for RecordingContainer class.
+   */
+  ~RecordingContainers();
+
+  /**
+   * @brief Adds a Container object to the container.
+   * @param data The Container object to add.
+   */
+  void addData(std::unique_ptr<Container> data);
```

Comment:
I would rename `addData` to `addContainer` (just for consistency with the other names e.g. `getContainer` and `RecordingContainers`)